client/app/views/menu_item.coffee

```diff
@@ -29,6 +25,16 @@ module.exports = class MenuItemView extends BaseView
     afterRender: ->
         @buildBadge @model.get 'color'
 
+    toggleVisible: ->
+        unless app.router.onCalendar
+            app.router.navigate 'calendar', true
+        @startSpinner()
+        setTimeout => # async, to allow spinner to show up.
+                @model.set 'visible', not @model.get 'visible'
+                @stopSpinner()
+                @render()
+            , 1
```

Comment:
Despite of my efforts, toggle visibility of calendars in list view on smartphone can last long (some seconds). The spinner is welcome as a feedback for the user in this case.